pkg/controller/autoimport/autoimport_controller.go

```diff
@@ -79,6 +81,16 @@ func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	// ensure the klusterlet manifest works exist
+	listOpts := &client.ListOptions{Namespace: managedClusterName}
+	manifestWorks := &workv1.ManifestWorkList{}
+	if err := r.client.List(ctx, manifestWorks, listOpts); err != nil {
+		return reconcile.Result{}, err
+	}
+	if !helpers.HasKlusterletManifestWorks(managedClusterName, manifestWorks) {
+		return reconcile.Result{}, fmt.Errorf("there are no klusterlet manifest works for managed cluster %s", managedClusterName)
```

Comment:
should we return error or nil? I think if we watch the work here, we can return nil here.

Reply:
I don't want to watch the work here :), the work is created along with the import-secret, we already watch the import-secret, in most conditions, this error will not happen, and we use the controller-runtime, it's hard to customize the work informer, so if we watch the work, I think it will bring extra cost

Reply:
use watching the manifest works instead of watching the import-secret